### ISSUE TITLE
PM-927

### ISSUE THREAD

**Issue by mayani** [pegasus-isi/pegasus#1044]
PM-927

**Comment by mayani** [comment]
**Author:** @juve

I'm thinking that we should change 4.5.1cvs to interpret "-r" as "Don't check kickstart invocation records, but check everything else". And we should change 4.6.0 to have a new flag that means "Don't check kickstart invocation records" and figure out what we should do with -r.

**Comment by mayani** [comment]
**Author:** @vahi

currently -r is set for mpi jobs ( that are not PMC ) and other jobs ( like create dir and deploy worker package jobs )in the worker package staging scenario. I agree that -r should be used in conjunction with the success and error messages if specified. 

the downside is the case, where the stdout/stderr of the job is huge and there is a problem transferring it back. The $RETURN is supposed to indicate that the job succeeded but something else went wrong, which is why we may have partial kickstart records etc.  Was that not the original reason for having -r option to exitcode , in addition to be able to rotate the stdout files in the case where kickstart is not used.

**Comment by mayani** [comment]
**Author:** @juve

My understanding was that -r should be used in all cases where Kickstart is not used.

In the case where stdout/stderr is not transferred properly, I don't think we can consider a job successful in that case, even if we have -r 0.

For right now, I think we should change it so that -r 0 skips the invocation record checks, but not the success message / failure message checks. If the value of -r is not zero, then we consider it a failure, otherwise we perform the other checks and let them determine the outcome.

I think that fits with our logic of "exitcode is looking for failures, not success".

We might also want to add another flag, like --check-invocations, for the Kickstart case to make it clear. And possibly a --check-cluster flag for clustered jobs and PMC. That way we can easily determine what is being checked.

For auxilliary jobs like create dir and worker package staging the planner should probably be adding success messages (esp. if they don't run locally). I believe that is already the case for PegasusLite jobs because I took out the PegasusLite-specific exitcode checks when we implemented success/failure messages.
